### ISSUE TITLE
SD cards Phison (PNY) SD256/SD512 fail to init by timing out in IF_COND cmd

### ISSUE THREAD

**Issue by LuisDennis** [CTCaer/hekate#520]
After updating hekate to v5.5.0 I get an error that says:
`Failed to mount SD!`
`Missing LP0 (sleep mode) lib!`
`Missing old Minerva lib!`

`Update bootloader folder!`

`Press any key...`

I already had the bootloader from v5.5.0 on my microSD Card. Injecting an older version of hekate like hekate_ctcaer_5.3.4.bin to my Nintendo Switch (even with the v5.5.0 bootloader folder on my microSD Card) works and using a different microSD Card with the hekate v5.5.0 bootloader folder works. I'm using this microSD Card https://i5.walmartimages.com/asr/5ad3ec8d-a9a4-4406-927b-e88054fb2ba0_1.f2a8686ba7492332a91cd6ff3a07c801.jpeg.

**Comment by CTCaer** [comment]
What sd card is that

Get a screenshot or photo from `Console info` -> `Sd card`.

**Comment by LuisDennis** [comment]
![IMG_1798](https://user-images.githubusercontent.com/66596892/101269460-c5207d80-3734-11eb-963f-62f325dc06d9.jpg)

**Comment by urherenow** [comment]
I wonder if when you right-click on the zip file and go to properties, if there is an "unblock" checkbox there, or if for some strange reason the archive bit is set on your bootloader folder.

**Comment by LuisDennis** [comment]
> I wonder if when you right-click on the zip file and go to properties, if there is an "unblock" checkbox there, or if for some strange reason the archive bit is set on your bootloader folder.

It was blocked, after unblocking it I retried with the files from the unblocked .zip and it still gave me the same error.

**Comment by CTCaer** [comment]
@LuisDennis dm me via retronx discord when you have time for testing.

**Comment by LuisDennis** [comment]
> @LuisDennis dm me via retronx discord when you have time for testing.

What's your Discord username? Mine is Loser Monkey#5556.

**Comment by CTCaer** [comment]
Try to force shut off (Press power for 12s). And then inject hekate. Tell me what happens.

**Comment by samwelnella** [comment]
I am having a similar issue. Using hekate 5.5.0 payload I get this error:
`Failed to mount SD!
Missing LP0 (sleep mode) lib!
Missing old Minerva lib!

Update bootloader folder!

Press any key... `

Using hekate 5.3.4 payload my SD card works fine.

[This](https://www.bestbuy.com/site/pny-pro-elite-512gb-microsdxc-uhs-i-memory-card/6329148.p?skuId=6329148) is the microSD card I’m using. And the SD card info is as follows:

![2F8F5B85-D277-4445-A0C9-D41CECAE660C](https://user-images.githubusercontent.com/1754179/101669029-5a668f00-3a1f-11eb-8bd6-fcb0ca567819.jpeg)

**Comment by CTCaer** [comment]
Join retronx discord and pm me. This can't be easily debugged here.

**Comment by CTCaer** [comment]
So @samwelnella helped me debug this and the issue is weird.
It seems to only happen on these PNY (Phison made) sd cards, based on the current 2 reports.

What causes it was identified, but needs investigation on the why.
Because it was something added that was making the driver more close to HOS one.

For now here are the 2 binaries that mitigate it:
[mitigation.zip](https://github.com/CTCaer/hekate/files/5669984/mitigation.zip)
nyx.bin goes into /bootloader/sys/.

@LuisDennis Please reply if that works for you also.

**Comment by acedaemon** [comment]
I get the same error message with 5.5.0 
The mitigation works except when using reboot to payload from the homebrew menu
And I have a SanDisk
![out](https://user-images.githubusercontent.com/31251652/101866118-a0b21000-3b3d-11eb-8352-f62a08923d5e.jpg)

**Comment by CTCaer** [comment]
@acedaemon You are probably describing another issue.
Does it work if you just reboot from the power menu?

**Comment by acedaemon** [comment]
Yes, that reboots to hekate. Do you want me to open another issue then?

**Comment by CTCaer** [comment]
@acedaemon That's an issue with the bundled r2p homebrew in atmosphere.

**Comment by acedaemon** [comment]
@CTCaer Alright. Thanks for the mitigation

**Comment by CTCaer** [comment]
@acedaemon 
You are welcome but the mitigation doesn't affect you.
The issue described by LuisDennis and samwelnella only happen on these Phison cards.

Your case can be fixed on a new atmosphere release. Otherwise you can recompile it yourself and changing 
https://github.com/Atmosphere-NX/Atmosphere/blob/master/troposphere/reboot_to_payload/source/main.c#L7
from `0x20000` to `0x24000`

**Comment by LuisDennis** [comment]
hekate v5.5.1 fixed the error.

**Comment by CTCaer** [comment]
The issue was extensively debugged but there's no solution.
Seems like an issue with the card's first init. hekate will continue to mitigate that.
